R/vendor.R

```diff
@@ -45,7 +45,7 @@ vendor <- function(version    = NULL,
   }
 
   # get renv sources
-  sources <- sources %||% renv_vendor_sources(version)
+  sources <- sources %||% renv_vendor_sources(version %||% renv_metadata_version())
```

Comment:
I think this now needs to be a commit SHA rather than a version, now that we're no longer tagging every renv commit. @hadley, what do you think the right approach is? If `version` is NULL, should we just use the SHA associated with the tip of main?

Reply:
With these changes, it used e7ec59ff4a92ec15583ac60faa1100f5b490af34 after I had installed this branch with remotes.

```r
renv:::renv_metadata_version()
#> [1] "e7ec59ff4a92ec15583ac60faa1100f5b490af34"
``` 

From the DESCRIPTION:

```
RemoteSha: e7ec59ff4a92ec15583ac60faa1100f5b490af34
```

There is definitely a question about how folks are going to use vendor... will they use vendor to incorporate the code for the version of renv installed locally, or should we require a remote reference? If we have a remote reference, then there is a risk of that remote version needing some alternate strategy for composing all the code.

I don't think that the vendor function can be used with old renv, for example.

I would prefer if `renv:::vendor` lost its inferred argument values; force us to provide values.

Reply:
IMO I think we should just default it to `main` — that's likely what you will want most of the time when re-vendoring anyway.  But I'm also fine with leaving version with no default so that you have to pick a sha yourself.

Reply:
I've made (version or sources) required.